File: pyg-manager-web/src/main/webapp/js/service/brandService.js
//定义service服务层
app.service("brandService", function ($http) {
    //把内置服务发送请求方法全部抽取到服务层代码
    //查询所有
    this.findAll = function () {
        return $http.get("../brand/findAll");
    };

    //分页查询
    this.findPage = function (page, rows) {
        return $http.get("../brand/findPage/" + page + "/" + rows);
    };

    //添加
    this.add = function (entity) {
        return $http.post("../brand/add", entity);
    };

    //修改
    this.update = function (entity) {
        return $http.post("../brand/update", entity);
    };
    //批量删除
    this.del = function (ids) {
        return $http.get("../brand/del/" + ids);
    };

    //根据id查询
    this.findOne = function (id) {

        return $http.get("../brand/findOne/" + id);

    }
    //查询品牌列表数据
    this.findBrandList = function () {
        return $http.get("../brand/findBrandList");
    }

});

File: pyg-manager/pyg-manager-interface/src/main/java/com/pyg/manager/service/BrandService.java
package com.pyg.manager.service;

import com.pyg.pojo.TbBrand;
import com.pyg.utils.PageResult;

import java.util.List;
import java.util.Map;

/**
 * Created by on 2018/8/4.
 */
public interface BrandService {

    /**
     * 需求：查询所有品牌数据
     */
    public List<TbBrand> findAll();

    /**
     * 需求：查询所有品牌数据，进行分页展示
     */
    public PageResult findPage(Integer page,Integer rows);
    /**
     * 需求：插入品牌数据
     * 返回值：void
     */
    public  void  add(TbBrand brand);

    /**
     * 需求：修改品牌数据
     * 返回值：void
     */
    public  void  update(TbBrand brand);

    /**
     * 需求：根据id查询品牌数据
     */
    public TbBrand findOne(Long id);

    /**
     * 需求：删除品牌数据
     * 返回值：void
     */
    public  void  delete(Long[] ids);

    /**
     * 需求:查询品牌下拉列表
     * 查询数据格式：[{id:'1',text:'联想'},{id:'2',text:'华为'}]
     * 返回值：List<Map>
     *
     */
    public List<Map> findBrandList();

}
